internal/ingress/controller/kong.go

```diff
@@ -72,10 +74,22 @@ func (n *KongController) OnUpdate(ctx context.Context, state *kongstate.KongStat
 	} else {
 		err = n.onUpdateDBMode(targetContent)
 	}
+	var target []byte
+	if n.cfg.DumpConfig == "enabled" {
+		target, _ = json.Marshal(targetContent)
+	} else if n.cfg.DumpConfig == "sensitive" {
+		sanitizedState := state.SanitizedCopy()
+		target, _ = json.Marshal(n.toDeckContent(ctx, sanitizedState))
+	}
 	if err != nil {
+		if len(target) > 0 {
+			_ = ioutil.WriteFile(filepath.Join(n.tmpDir, "target.json"), target, 0600)
+			_ = ioutil.WriteFile(filepath.Join(n.tmpDir, "last_good.json"), n.lastConfig, 0600)
+		}
 		return err
 	}
 	n.runningConfigHash = shaSum
+	n.lastConfig = target
```

Comment:
Would this way to write "last good config" and "last bad config" work as well:

1. try to apply config
2. if application was successful, write to "last_good.json"
3. otherwise, write to "last_failed.json"?

This would free us of the need to remember `lastConfig`.